www/docs/deprecations.md

```diff
@@ -41,6 +41,157 @@ PS: Don't forget to add it to cmd/mcp.go as well!
+          - "--label=org.opencontainers.image.version={{.Version}}"
+          - "--label=org.opencontainers.image.source={{.GitURL}}"
+          - "--platform=linux/arm64"
+        goarch: arm64
+
+    docker_manifests:
+      - name_template: "foo/bar:v{{ .Version }}"
+        image_templates:
+          - "foo/bar:v{{ .Version }}-amd64"
+          - "foo/bar:v{{ .Version }}-arm64"
+      - name_template: "ghcr.io/foo/bar:v{{ .Version }}"
+        image_templates:
+          - "ghcr.io/foo/bar:v{{ .Version }}-amd64"
+          - "ghcr.io/foo/bar:v{{ .Version }}-arm64"
+      - name_template: "{{ if not .IsNightly }}foo/bar:latest{{ end }}"
+        image_templates:
+          - "foo/bar:v{{ .Version }}-amd64"
+          - "foo/bar:v{{ .Version }}-arm64"
+      - name_template: "{{ if not .IsNightly }}ghcr.io/foo/bar:latest{{ end }}"
+        image_templates:
+          - "ghcr.io/foo/bar:v{{ .Version }}-amd64"
+          - "ghcr.io/foo/bar:v{{ .Version }}-arm64"
+      - name_template: "{{ if .IsNightly }}foo/bar:nightly{{ end }}"
+        image_templates:
+          - "foo/bar:v{{ .Version }}-amd64"
+          - "foo/bar:v{{ .Version }}-arm64"
+      - name_template: "{{ if .IsNightly }}ghcr.io/foo/bar:nightly{{ end }}"
+        image_templates:
+          - "ghcr.io/foo/bar:v{{ .Version }}-amd64"
+          - "ghcr.io/foo/bar:v{{ .Version }}-arm64"
+    ```
+
+=== "After"
+
+    ```yaml
+    dockers_v2:
+      - images:
+          - "foo/bar"
+          - "ghcr.io/foo/bar"
+        tags:
+          - "v{{ .Version }}"
+          - "{{ if .IsNightly }}nightly{{ end }}"
+          - "{{ if not .IsNightly }}latest{{ end }}"
+        labels:
+          "org.opencontainers.image.description": "Foo bar"
+          "org.opencontainers.image.created": "{{.Date}}"
+          "org.opencontainers.image.name": "{{.ProjectName}}"
+          "org.opencontainers.image.revision": "{{.FullCommit}}"
+          "org.opencontainers.image.version": "{{.Version}}"
+          "org.opencontainers.image.source": "{{.GitURL}}"
+    ```
+
+As you can see, it's a lot simpler.
+The resulting images are the same, a combination of all the non-empty images
+with all the non-empty tags.
+
+This will also require a small change in your `Dockerfile` when copying from the
+context:
+
+=== "Before"
+
+    ```dockerfile
+    FROM alpine
+    COPY my-binary /usr/bin
+    ```
+
+=== "After"
+
+    ```dockerfile
+    FROM alpine
+    ARG TARGETPLATFORM
+    COPY $TARGETPLATFORM/my-binary /usr/bin
+    ```
+
+GoReleaser will automatically setup the context in such a way that all the
+artifacts for the given target platform will be located within
+`$TARGETPLATFORM/`.
+
+One side effect of this new feature is that builds with `--snapshot` will
+produce images that cannot be ran - it only tests that the `docker build`
+succeeds.
+This is something we'll need to improve in future versions.
+
+Feel free to suggest improvements
+[here](https://github.com/orgs/goreleaser/discussions/6005).
```

Comment:
What about having a push option, which has the value `push: {{ not .IsSnapshot }}` by default. If an end-user really wants to have usage images from a snapshot build, he can opt-in. He also can use templated `image_templates` to configure different push location.

My use case here: I'm running `--snapshot` on main branches and I would like to push the main branch to the `main` tag. Thats one way, how I can quickly get feedback from reporters without publish a full release.